fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/ManageElectionFragment.java

```diff
@@ -75,43 +113,229 @@ public View onCreateView(
+        editNameBuilder.setTitle("Edit Election Name");
+        editNameBuilder.setMessage("Please enter the new name you want for the election ");
+// Set up the input
+        final EditText inputName = new EditText(getContext());
+
+        inputName.setHint("New Name");
+
+
+// Set up the buttons
+        editNameBuilder.setPositiveButton("SUBMIT", (dialog, which) -> {
+            newName = inputName.getText().toString();
+            requestCode = EDIT_NAME_CODE;
+            builder.show();
+        });
+        editNameBuilder.setNegativeButton("CANCEL", (dialog, which) -> {
+            dialog.cancel();
+        });
+
+        editNameBuilder.create();
+
+        // create the Alert Dialog to edit question
+        AlertDialog.Builder editQuestionBuilder = new AlertDialog.Builder(getContext());
+        editQuestionBuilder.setTitle("Edit Election Question");
+        editQuestionBuilder.setMessage("Please enter the new question you want for the election ");
+// Set up the input
+        final EditText inputQuestion = new EditText(getContext());
+
+        inputQuestion.setHint("New Question");
+
+
+// Set up the buttons
+        editQuestionBuilder.setPositiveButton("SUBMIT", (dialog, which) -> {
+            newQuestion = inputQuestion.getText().toString();
+            requestCode = EDIT_QUESTION_CODE;
+            builder.show();
+        });
+        editQuestionBuilder.setNegativeButton("CANCEL", (dialog, which) -> {
+            dialog.cancel();
+        });
+
+        editQuestionBuilder.create();
+
+        // On click, edit new name button
+
+        editName.setOnClickListener(v -> {
+            inputName.setText(null); // we make sure the text is blank when we reclick the button
+            if (inputName.getParent() != null) {
+                ((ViewGroup) inputName.getParent()).removeView(inputName);
+            }
+            editNameBuilder.setView(inputName);
+
+            editNameBuilder.show();
+        });
+
+        // On click, edit new question button
+
+        editQuestion.setOnClickListener(v -> {
+            inputQuestion.setText(null); // we make sure the text is blank when we click the button
+            if (inputQuestion.getParent() != null) {
+                ((ViewGroup) inputQuestion.getParent()).removeView(inputQuestion);
+            }
+            editQuestionBuilder.setView(inputQuestion);
+            editQuestionBuilder.show();
+        });
+
+        // On click, edit start time button
+        editStartTimeButton.setOnClickListener(
                 v -> {
-                   // TODO implement the cancel election action with backend when the backend is ready
-                    laoDetailViewModel.terminateCurrentElection();
-                    laoDetailViewModel.openLaoDetail();
+                    // we set the request code
+                    requestCode = START_TIME_CODE;
+                    // show the timePicker
+                    timePickerDialog.show();
```

Comment:
These four lines are used multiple times, might be a good idea to refactor and create a method that takes the requestCode as an argument and then implements those lines